Tests are failing on Solaris

Solaris place krb5 headers into /usr/include/kerberosv5/ so
the -I/usr/include/kerberosv5 option need to be passed into the
test by pointing to "lib" entity from where it can be retrieved.

--- a/wscript_configure_system_mitkrb5	2018-06-26 07:42:46.000000000 +0000
+++ b/wscript_configure_system_mitkrb5	2018-08-06 06:54:35.736827959 +0000
@@ -75,8 +75,8 @@
             Logs.info('MIT Kerberos %s detected, MIT krb5 build can proceed' % (krb5_version))
 
 conf.CHECK_CFG(args="--cflags --libs", package="com_err", uselib_store="com_err")
-conf.CHECK_FUNCS_IN('_et_list', 'com_err')
-conf.CHECK_HEADERS('com_err.h', lib='com_err')
+conf.CHECK_FUNCS_IN('_et_list', 'com_err krb5')
+conf.CHECK_HEADERS('com_err.h', lib='com_err krb5')
 conf.CHECK_HEADERS('kdb.h', lib='kdb5')
 
 conf.CHECK_HEADERS('krb5.h krb5/locate_plugin.h', lib='krb5')
@@ -84,12 +84,12 @@
 possible_gssapi_headers="gssapi.h gssapi/gssapi_generic.h gssapi/gssapi.h gssapi/gssapi_ext.h gssapi/gssapi_krb5.h gssapi/gssapi_oid.h"
 conf.CHECK_HEADERS(possible_gssapi_headers, lib='gssapi')
 
-conf.CHECK_FUNCS_IN('krb5_encrypt_data', 'k5crypto')
-conf.CHECK_FUNCS_IN('des_set_key','crypto')
-conf.CHECK_FUNCS_IN('copy_Authenticator', 'asn1')
-conf.CHECK_FUNCS_IN('roken_getaddrinfo_hostspec', 'roken')
+conf.CHECK_FUNCS_IN('krb5_encrypt_data', 'k5crypto krb5')
+conf.CHECK_FUNCS_IN('des_set_key','crypto krb5')
+conf.CHECK_FUNCS_IN('copy_Authenticator', 'asn1 krb5')
+conf.CHECK_FUNCS_IN('roken_getaddrinfo_hostspec', 'roken krb5')
 
-conf.CHECK_HEADERS('profile.h')
+conf.CHECK_HEADERS('profile.h', lib='krb5')
 
 if conf.CHECK_FUNCS_IN('gss_display_status', 'gssapi gssapi_krb5'):
     have_gssapi=True
@@ -111,7 +111,7 @@
        gss_import_cred gss_export_cred
        gss_acquire_cred_from
        ''', 'gssapi gssapi_krb5')
-conf.CHECK_VARIABLE('GSS_KRB5_CRED_NO_CI_FLAGS_X', headers=possible_gssapi_headers)
+conf.CHECK_VARIABLE('GSS_KRB5_CRED_NO_CI_FLAGS_X', headers=possible_gssapi_headers, lib='krb5')
 conf.CHECK_FUNCS_IN('krb5_mk_req_extended krb5_kt_compare', 'krb5')
 conf.CHECK_FUNCS('''
        krb5_auth_con_getrecvsubkey
@@ -149,25 +149,25 @@
 conf.CHECK_DECLS('''krb5_get_credentials_for_user
                     krb5_auth_con_set_req_cksumtype''',
                     headers='krb5.h', always=True)
-conf.CHECK_VARIABLE('AP_OPTS_USE_SUBKEY', headers='krb5.h')
-conf.CHECK_VARIABLE('KV5M_KEYTAB', headers='krb5.h')
-conf.CHECK_VARIABLE('KRB5_KU_OTHER_CKSUM', headers='krb5.h')
-conf.CHECK_VARIABLE('KRB5_KEYUSAGE_APP_DATA_CKSUM', headers='krb5.h')
-conf.CHECK_VARIABLE('ENCTYPE_AES128_CTS_HMAC_SHA1_96', headers='krb5.h')
-conf.CHECK_VARIABLE('ENCTYPE_AES256_CTS_HMAC_SHA1_96', headers='krb5.h')
+conf.CHECK_VARIABLE('AP_OPTS_USE_SUBKEY', headers='krb5.h', lib='krb5')
+conf.CHECK_VARIABLE('KV5M_KEYTAB', headers='krb5.h', lib='krb5')
+conf.CHECK_VARIABLE('KRB5_KU_OTHER_CKSUM', headers='krb5.h', lib='krb5')
+conf.CHECK_VARIABLE('KRB5_KEYUSAGE_APP_DATA_CKSUM', headers='krb5.h', lib='krb5')
+conf.CHECK_VARIABLE('ENCTYPE_AES128_CTS_HMAC_SHA1_96', headers='krb5.h', lib='krb5')
+conf.CHECK_VARIABLE('ENCTYPE_AES256_CTS_HMAC_SHA1_96', headers='krb5.h', lib='krb5')
 conf.CHECK_DECLS('KRB5_PDU_NONE', reverse=True, headers='krb5.h')
 conf.CHECK_STRUCTURE_MEMBER('krb5_keytab_entry', 'key', headers='krb5.h',
-                            define='HAVE_KRB5_KEYTAB_ENTRY_KEY')
+                            define='HAVE_KRB5_KEYTAB_ENTRY_KEY', lib='krb5')
 conf.CHECK_STRUCTURE_MEMBER('krb5_keytab_entry', 'keyblock', headers='krb5.h',
-                            define='HAVE_KRB5_KEYTAB_ENTRY_KEYBLOCK')
+                            define='HAVE_KRB5_KEYTAB_ENTRY_KEYBLOCK', lib='krb5')
 conf.CHECK_STRUCTURE_MEMBER('krb5_address', 'magic', headers='krb5.h',
-                            define='HAVE_MAGIC_IN_KRB5_ADDRESS')
+                            define='HAVE_MAGIC_IN_KRB5_ADDRESS', lib='krb5')
 conf.CHECK_STRUCTURE_MEMBER('krb5_address', 'addrtype', headers='krb5.h',
-                            define='HAVE_ADDRTYPE_IN_KRB5_ADDRESS')
+                            define='HAVE_ADDRTYPE_IN_KRB5_ADDRESS', lib='krb5')
 conf.CHECK_STRUCTURE_MEMBER('krb5_ap_req', 'ticket', headers='krb5.h',
-                            define='HAVE_TICKET_POINTER_IN_KRB5_AP_REQ')
+                            define='HAVE_TICKET_POINTER_IN_KRB5_AP_REQ', lib='krb5')
 conf.CHECK_STRUCTURE_MEMBER('krb5_prompt', 'type', headers='krb5.h',
-                            define='HAVE_KRB5_PROMPT_TYPE')
+                            define='HAVE_KRB5_PROMPT_TYPE', lib='krb5')
 
 conf.CHECK_TYPE('krb5_encrypt_block', headers='krb5.h')
 
@@ -177,7 +177,7 @@
        krb5_get_init_creds_opt_free(ctx, opt);
        ''',
     'KRB5_CREDS_OPT_FREE_REQUIRES_CONTEXT',
-    headers='krb5.h', link=False,
+    headers='krb5.h', link=False, lib='krb5',
     msg="Checking whether krb5_get_init_creds_opt_free takes a context argument")
 conf.CHECK_CODE('''
        const krb5_data *pkdata;
@@ -288,7 +288,7 @@
 conf.CHECK_CODE('''#define KRB5_DEPRECATED 1
        #include <krb5.h>''',
    'HAVE_KRB5_DEPRECATED_WITH_IDENTIFIER', addmain=False,
-    link=False,
+    link=False, lib='krb5',
     msg="Checking for KRB5_DEPRECATED define taking an identifier")
 
 conf.CHECK_CODE('''
